httphandlers/http_handlers.go

```diff
@@ -43,29 +43,34 @@ func (h httpMetricsHandler) ServeHTTP(w http.ResponseWriter, req *http.Request)
 
 type handlerOpt func(h *transactionAwareRequestLoggingHandler)
 
+// HeaderFilterCallback is a callback type for additional header filtering when logging request headers
+// The Callback is provided the header key as string and it's expected to return true if the header needs to be removed.
+type HeaderFilterCallback func(key string) bool
```

Comment:
`Callback` is not good choice for the name, since it does not provide useful information regarding the type. `HeaderFilter` will be fine.